src/ape/managers/accounts.py

```diff
@@ -1,12 +1,81 @@
-from typing import Dict, Iterator, List, Type
+from typing import Dict, Iterator, List, Optional, Type
 
 from ape.api.accounts import AccountAPI, AccountContainerAPI, TestAccountAPI
-from ape.types import AddressType
-from ape.utils import cached_property, singledispatchmethod
+from ape.api.providers import TransactionAPI
+from ape.types import AddressType, MessageSignature, SignableMessage, TransactionSignature
+from ape.utils import ManagerAccessMixin, cached_property, singledispatchmethod
 
 from .base import BaseManager
 
 
+class ImpersonatedAccount(AccountAPI):
+    raw_address: AddressType
+
+    @property
+    def address(self) -> AddressType:
+        return self.raw_address
+
+    def sign_message(self, msg: SignableMessage) -> Optional[MessageSignature]:
+        raise NotImplementedError("This account cannot sign messages")
+
+    def sign_transaction(self, txn: TransactionAPI) -> Optional[TransactionSignature]:
+        return None
+
+
+class TestAccountContainer(list, ManagerAccessMixin):
+    @property
+    def accounts(self) -> Iterator[AccountAPI]:
+        for plugin_name, (container_type, account_type) in self.plugin_manager.account_types:
+            if not issubclass(account_type, TestAccountAPI):
+                continue
+
+            # pydantic validation won't allow passing None for data_folder/required attr
+            container = container_type(data_folder="", account_type=account_type)
+            for account in container.accounts:
+                yield account
+
+    def aliases(self) -> Iterator[str]:
+        for account in self.accounts:
+            if account.alias:
+                yield account.alias
+
+    def __len__(self) -> int:
+        return len(list(self.accounts))
+
+    @singledispatchmethod
+    def __getitem__(self, account_id):
+        raise NotImplementedError(f"Cannot use {type(account_id)} as account ID.")
+
+    @__getitem__.register
+    def __getitem_int(self, account_id: int):
+        for idx, account in enumerate(self.accounts):
+            if account_id == idx:
+                return account
+
+        raise IndexError(f"No account at index '{account_id}'.")
+
+    @__getitem__.register
+    def __getitem_str(self, account_str: str):
```

Comment:
It could be on another PR, but we could support aliases here so you can load our own accounts in a test environment and automatically have funding.